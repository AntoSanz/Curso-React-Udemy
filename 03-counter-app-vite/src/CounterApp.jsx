import { useState } from 'react'; //Importar hook useState. Los hook en react siempre empiezan por use.
import PropTypes from 'prop-types';

export const CounterApp = ({ value }) => {
    console.log('render');
    
    const [counter, setCounter] = useState(value); //Primer valor es el valor que declaramos en useState(AQUI) y el segundo el metodo que se va a utilizar para modificar ese valor
    
    const handleAdd = () => {
        setCounter((c) => c + 1);  
    };

    const handleSubstract = () => {
        setCounter(counter - 1);
    };

    const handleReset = () => {
        setCounter(value);
    };

    return (
        <>
            <h1>CounterApp</h1>
            <h2> { counter } </h2>
            <button onClick={ handleAdd }> +1 </button>
            <button onClick={ handleSubstract }> -1 </button>
            <button onClick={ handleReset }> Reset </button>
        </>
    );
};

CounterApp.propTypes = {
    value: PropTypes.number.isRequired,
}
CounterApp.defaultProps = {
    value: 0,
}

