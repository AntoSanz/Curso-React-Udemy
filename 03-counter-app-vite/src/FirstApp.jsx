// Si no hay dependencias se recomienda que las funciones, constantes, variables estÃ©n fuera del componente
const newString = "Hello World!";
const newArray = [1, 2, 3, 4, 5, 6, 7, 8, 9];
const newObject = {
  nombre: "Antonio",
  apellido: "Sanz",
};
const getString = () => {
  return "Hello function!";
};

export const FirstApp = () => {
  return (
    <>
      {/* <></> es lo mismo que <Fragment></Fragment> y no es necesario importar el Fragment*/}
      <h2>{newString}</h2>
      <p>{newArray}</p>
      {/* React no puede procesar objetos de manera que para mostrar el objeto tenemos que usar una expresion de JS: JSON.stringify(Objeto)
            Para mostar las porpiedades del objeto podemos llamar Objeto.propiedad */}
      <code>{JSON.stringify(newObject)}</code>
      <p>{getString()}</p>
    </>
  );
};
