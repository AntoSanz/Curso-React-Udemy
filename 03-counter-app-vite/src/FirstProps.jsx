//rafc = Short cut para crear un componente
import PropTypes from 'prop-types';

export const FirstProps = ({title, age, name}) => {
    return(
        <>  
            <h1 data-testid="test-title">{title}</h1>
            <p>{name}</p>
            <p>Edad: {age}</p>
        </>
    );
}

FirstProps.propTypes = {
    subTitle: PropTypes.number, //Propiedad de tipo number no obligatoria
    name: PropTypes.string, //Propiedad de tipo string y obligatoria
    title: PropTypes.string.isRequired,
}

FirstProps.defaultProps = {
    // title: 'No hay ningún título.',
}