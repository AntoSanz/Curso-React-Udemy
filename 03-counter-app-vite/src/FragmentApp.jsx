import { Fragment } from 'react'
export const FragmentApp = () => {
  return (
    <Fragment>
      <div>Hola Mundo!</div>
      <p>Soy un subtitulo</p>
    </Fragment>
  )
}

//Si no se quiere importar el fragmento solo hay qu emeter los elementos entre <> </> en vez de importar el Fragment y meterlos entre <Fragment></Fragment>

// export const FragmentApp = () => {
//   return (
//     <> 
//       <div>Div dentro del Fragment</div>
//       <p>Subtitulo</p>
//     </>
//   )
// }