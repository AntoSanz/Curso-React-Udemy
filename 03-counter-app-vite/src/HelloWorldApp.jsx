//Autocompletar para crear un componente se puede utilizar rafc
export const HelloWorldApp = () => {
  return (
    <h1>Hello World App</h1>
  )
}
