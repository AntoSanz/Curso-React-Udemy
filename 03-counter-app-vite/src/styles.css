html, body {
    /* Ordenar por orden alfabetico: CTRL + SHIFT + P > Ordenar */
    background-color: #21232A;
    color: white;
    font-family: Helvetica, Arial, sans-serif;
    font-size: 1.3rem;
    padding: 70px; 
}