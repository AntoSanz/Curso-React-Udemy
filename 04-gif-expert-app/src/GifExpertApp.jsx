import { useState } from 'react';

export const GifExpertApp = () => {
    //useStateSnippet
    const [categories, setCategories] = useState(['One Punch', 'Dragon Ball']);

    const onAddCategory = () => {
        let newCategory = 'One Piece';
        // setCategories(c => [c, newCategory]);
        setCategories([...categories, newCategory]);
    };

    return (
        <>
            <h1>GifExpertApp</h1>
            <button onClick={onAddCategory}>AÃ±adir</button>
            <ol>
                {categories.map((category) => {
                    return <li key={category}>{category}</li>;
                })}
            </ol>
        </>
    );
};
