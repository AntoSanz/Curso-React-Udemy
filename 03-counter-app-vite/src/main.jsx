import React from "react";
import ReactDOM from "react-dom/client";
// import { HelloWorldApp } from "./helloWorldApp";
import { FirstApp } from "./FirstApp";
import "./styles.css";
import { FirstProps } from "./FirstProps";

ReactDOM.createRoot(document.getElementById("root")).render(
    <React.StrictMode>
        {/* <HelloWorldApp /> */}
        <FirstApp />
        {/* <FirstProps title="Hola soy Anto" subTitle={123} /> */}
        <FirstProps subTitle={123} title="Hello props" />
    </React.StrictMode>
);
