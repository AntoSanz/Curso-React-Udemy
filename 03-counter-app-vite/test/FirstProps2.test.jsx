import { render, screen } from '@testing-library/react';
import { FirstProps } from '../src/FirstProps';

describe('Pruebas en <FirstProps /> ', () => {
    const title = 'Hello World';
    const name = 'Antonio Sanz';

    test('Debe coincidir con el snapshot', () => {
        const { container } = render(<FirstProps title={title} />);
        expect(container).toMatchSnapshot();
    });

    test('Debe mostrar el texto "Hello World"', () => {
        render(<FirstProps title={title} />);
        expect(screen.getByText(title)).toBeTruthy();
    });

    test('Debe de mostrar el titulo en un <h1></h1>', () => {
        render(<FirstProps title={title} />);
        expect(screen.getByRole('heading', { level: 1 }).innerHTML).toContain(title);
    });

    test('Debe de mostrar el "name" enviado por props', () => {
        render(<FirstProps title={title} name={name} />);
        expect(screen.getAllByText(name)).toBeTruthy();
    });
});
