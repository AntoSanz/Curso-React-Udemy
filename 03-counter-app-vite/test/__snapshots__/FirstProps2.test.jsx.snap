// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <FirstProps />  Debe coincidir con el snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
    Hello World
  </h1>
  <p />
  <p>
    Edad: 
  </p>
</div>
`;
